### ISSUE TITLE
nss build failing

### ISSUE THREAD

**Issue by oliverchang** [google/oss-fuzz#858]
See: https://oss-fuzz-build-logs.storage.googleapis.com/log-8e29de20-b393-4f32-a417-546aff9eee35.txt

```
Step #3: In file included from ../../fuzz/tls_client_target.cc:7:
Step #3: /usr/local/bin/../include/c++/v1/memory:2233:5: error: delete called on non-final 'DummyPrSocket' that has virtual functions but non-virtual destructor [-Werror,-Wdelete-non-virtual-dtor]
Step #3:     delete __ptr;
Step #3:     ^
Step #3: /usr/local/bin/../include/c++/v1/memory:2546:7: note: in instantiation of member function 'std::__1::default_delete<DummyPrSocket>::operator()' requested here
Step #3:       __ptr_.second()(__tmp);
Step #3:       ^
Step #3: /usr/local/bin/../include/c++/v1/memory:2500:19: note: in instantiation of member function 'std::__1::unique_ptr<DummyPrSocket, std::__1::default_delete<DummyPrSocket> >::reset' requested here
Step #3:   ~unique_ptr() { reset(); }
Step #3:                   ^
Step #3: ../../fuzz/tls_client_target.cc:103:34: note: in instantiation of member function 'std::__1::unique_ptr<DummyPrSocket, std::__1::default_delete<DummyPrSocket> >::~unique_ptr' requested here
Step #3:   std::unique_ptr<DummyPrSocket> socket(new DummyPrSocket(data, len));
```

@ttaubert mind taking a look?

**Comment by ttaubert** [comment]
We didn't touch that code in a while. Was clang updated recently? Also, the attached log doesn't seem to contain the above snippet... looks like it's an ICU rather than an NSS build.

**Comment by oliverchang** [comment]
Sorry, mispaste. Here is the correct log: https://oss-fuzz-build-logs.storage.googleapis.com/log-9242d45d-a57b-44bb-920a-a7f0e310361f.txt 

We do regularly update our clang version, so this may be a new warning/error that was introduced. Would this be easy to fix?

**Comment by ttaubert** [comment]
Yeah, that should be very easy to fix. Working on a patch.

**Comment by ttaubert** [comment]
Should be fixed by cset [2564805e0d53](https://hg.mozilla.org/projects/nss/rev/2564805e0d53).

**Comment by oliverchang** [comment]
Yep, latest build succeeded. Thanks a lot for fixing!
